src/test/java/org/usrv/http/ServerTests.java

```diff
@@ -301,6 +301,22 @@ void serverHandlesMultipleRequests() throws Exception {
         }
     }
 
+    @Test
+    @DisplayName("Server should respond with headers for HEAD requests")
+    void serverRespondsWithHeadersForFile() throws Exception {
+        Files.writeString(Path.of(defaultDistDirectory.toString(), "newfile.txt"), "New content");
+
+        HttpRequest request = HttpRequest.newBuilder()
+                .uri(URI.create("http://localhost:80/newfile.txt"))
+                .HEAD()
+                .build();
+
+        HttpResponse<String> response = httpClient.send(request,
+                HttpResponse.BodyHandlers.ofString());
+
+        assertEquals(200, response.statusCode());
+        assertEquals("", response.body());
+    }
```

Comment:
This test contains a subtle bug. If you include a body as the response to a HEAD request, this will still pass since the HTTP lib ignores any body included as the response to a HEAD request. I would suggest adding a test using printwriter/bufferedreader like in `serverRespondsWith400()` in order to have more control, and allow any present body to be detected.